### PR TITLE
Update Sample File Upload and upgrade buttons to Shadcn button

### LINKED ISSUE #9153: No Option to Clear The Selected Image On Sample Test Page
**Describe the bug**
There's no option to clear the selected image on images uploaded in sample test page

**To Reproduce**

1. Go to '/sample' page
2. Click on **Update Sample Test Status** on any test card
3. Select test completed in New Status Field
4. Upload the image
5. See the error

**Expected behavior**
An option to clear the selected image should be provided

**Screenshots**

https://github.com/user-attachments/assets/5b3afd37-86d9-4bae-9ca3-580ef02f4fd1

**Desktop (please complete the following information):**

- OS: Windows
- Browser Chrome
- Version Latest

### PR DESCRIPTION
**Pull request by nithish1018**
## Proposed Changes

- Fixes #9153 
- Updated File Upload process to match with existing FileUpload
- Updated regular Button and ButtonV2 to Shadcn button

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Introduced a new file upload mechanism in the UpdateStatusDialog, enhancing user experience with improved error handling and progress indication.
	- Added a new label for report uploads in the localization files.
	- Introduced a new `primary` button variant for enhanced visual options.

- **Bug Fixes**
	- Restructured file upload handling to ensure smoother functionality and user interaction.

### CODE REVIEW

File: src/components/Patient/UpdateStatusDialog.tsx [hunk 1]
```diff
@@ -236,10 +236,17 @@ const UpdateStatusDialog = (props: Props) => {
...
+                    icon="l-times"
+                    className="text-lg cursor-pointer mt-2 mr-4"
+                    onClick={removeFile}
+                  />
+                )}
                 <Submit
                   type="submit"
                   onClick={handleUpload}
-                  disabled={uploadDone}
+                  disabled={file ? false : true}
```
[the remark this answers is not in the document]
Comment (nithishsiliveru) [reply src/components/Patient/UpdateStatusDialog.tsx]: Done

**Review by shivankacker — CHANGES_REQUESTED** [review]
@nithish1018 we need to migrate the entire flow to use the new `useFileUpload` and `useFileManager` hooks. Look at consent and file upload component for reference.

**Comment by nihal467** [comment]
check the code rabbit suggestion

**Comment by nithish1018** [comment]
> check the code rabbit suggestion

Done

File: src/components/Patient/UpdateStatusDialog.tsx [hunk 2]
```diff
@@ -10,31 +10,24 @@ import CheckBoxFormField from "@/components/Form/FormFields/CheckBoxFormField";
...
 import * as Notification from "@/Utils/Notifications";
-import routes from "@/Utils/request/api";
-import request from "@/Utils/request/request";
-import uploadFile from "@/Utils/request/uploadFile";
 
 interface Props {
-  sample: SampleTestModel;
-  handleOk: (sample: SampleTestModel, status: number, result: number) => void;
+  sample: any;
+  handleOk: (sample: any, status: number, result: number) => void;
```
[the remark this answers is not in the document]
Comment (nithishsiliveru) [reply src/components/Patient/UpdateStatusDialog.tsx]: Added, thank you for the suggestion

**Review by shivankacker — APPROVED** [review]

**Comment by nihal467** [comment]
@nithish1018 The design looks inconsistent compared to our file upload component in the screenshot below. Please implement a similar design for consistency. 

image

**Comment by nithish1018** [comment]
> @nithish1018 The design looks inconsistent compared to our file upload component in the screenshot below. Please implement a similar design for consistency.
> 
> image

I made the requested changes, please review it

File: src/components/Patient/UpdateStatusDialog.tsx [hunk 3]
```diff
@@ -218,36 +176,78 @@ const UpdateStatusDialog = (props: Props) => {
...
+                    <label
+                      key={i}
+                      className="flex w-full cursor-pointer items-center justify-center gap-2 rounded-lg border border-dashed border-primary-500/20 bg-primary-500/10 p-3 text-primary-700 transition-all hover:bg-primary-500/20 md:p-6"
+                      onClick={button.onClick}
+                      id={button.id}
+                    >
+                      <CareIcon icon={button.icon} className="text-2xl" />
+                      <div className="text-lg">{button.name}</div>
+                      {button.children}
+                    </label>
```
[the remark this answers is not in the document]
Comment (nithishsiliveru) [reply src/components/Patient/UpdateStatusDialog.tsx]: The existing FileUpload component doesn't have such implementation

File: src/components/Patient/UpdateStatusDialog.tsx [hunk 4]
```diff
@@ -218,36 +176,78 @@ const UpdateStatusDialog = (props: Props) => {
...
+                  label={t("enter_file_name")}
+                  id="upload-file-name"
+                  value={fileUpload.fileNames[0] || ""}
+                  disabled={fileUpload.uploading}
+                  onChange={(e) => fileUpload.setFileName(e.value)}
+                  error={fileUpload.error || undefined}
+                  required
+                />
+                <div className="flex items-center gap-2">
+                  <ButtonV2
```
Comment (Jacobjeevan) [inline src/components/Patient/UpdateStatusDialog.tsx]: Replace instances of button and ButtonV2 with shadcn's Button (from components/ui); we are moving to using shadcn for newer items 👍

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

File: src/components/Patient/UpdateStatusDialog.tsx (hunk 4 above)
Comment (nithishsiliveru) [reply src/components/Patient/UpdateStatusDialog.tsx]: Sure, will update it

**Review by Jacobjeevan — APPROVED** [review]

**Comment by nihal467** [comment]
image

- [ ] Only keep the discard button at the bottom and remove the other button next to the file name to maintain consistency with the existing file upload.

@nithish1018 make sure to do a basic QA before requesting a review

**Comment by nithish1018** [comment]
> image
> 
> * [x]  Only keep the discard button at the bottom and remove the other button next to the file name to maintain consistency with the existing file upload.
> 
> @nithish1018 make sure to do a basic QA before requesting a review

Actually the existing File Upload has that extra option to clear image , so I've added that here aswell.

image

Now I've updated the PR as per your suggestions, please review it

File: src/components/ui/button.tsx [hunk 5]
```diff
@@ -15,6 +15,8 @@ const buttonVariants = cva(
           "bg-red-500 text-gray-50 shadow-sm hover:bg-red-500/90 dark:bg-red-900 dark:text-gray-50 dark:hover:bg-red-900/90",
         outline:
           "border border-gray-200 bg-white shadow-sm hover:bg-gray-100 hover:text-gray-900 dark:border-gray-800 dark:bg-gray-950 dark:hover:bg-gray-800 dark:hover:text-gray-50",
+        primary:
+          "bg-primary-500 text-white shadow hover:bg-primary-700/90 dark:bg-primary-100 dark:text-primary-900 dark:hover:bg-primary-100/90",
```
Comment (rithviknishad) [inline src/components/ui/button.tsx]: As per figma designs, primary variant would be using shade 700.

image

https://www.figma.com/design/Z93EYKSa1MdBmXndsMBJSQ/Care?node-id=1285-2299&node-type=canvas&t=li0D8fzrfJMTag7g-0

```suggestion
          "bg-primary-700 text-white shadow hover:bg-primary-700/90 dark:bg-primary-100 dark:text-primary-900 dark:hover:bg-primary-100/90",
```

Comment (nithishsiliveru) [reply src/components/ui/button.tsx]: I'll update it

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

File: src/components/Patient/UpdateStatusDialog.tsx [hunk 6]
```diff
@@ -65,11 +61,27 @@ const UpdateStatusDialog = (props: Props) => {
...
+    show?: boolean;
+    id: string;
+  }[] = [
+    {
+      name: t("choose_file"),
+      icon: "l-file-upload-alt",
+      children: <fileUpload.Input />,
+      id: "upload-file",
+    },
+  ];
```
Comment (rithviknishad) [inline src/components/Patient/UpdateStatusDialog.tsx]: I don't see any places where the `uploadButtons` array is mutated. And it only contains 1 element. Why is it an array? Can't we just directly inline the component?

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/components/Patient/UpdateStatusDialog.tsx (hunk 6 above)
Comment (nithishsiliveru) [reply src/components/Patient/UpdateStatusDialog.tsx]: Actually I tried to rewrite this component just like the  File Upload component, Now I've pushed the requested changes. Please update the review

**Review by rithviknishad — APPROVED** [review]

**Pull request merged**